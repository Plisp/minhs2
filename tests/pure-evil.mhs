main = 
    let id :: Int -> Int x = x;
    in letrec id2 y = id y;
                id3 x = id2 x;
                two x y = one y x;
                one :: (Int -> Bool -> Int) x y = id x;
        in id2;
