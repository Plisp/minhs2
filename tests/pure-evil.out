main :: (Int -> Int) = let id :: (Int -> Int) x = x;
                       in letrec id2 :: (Int -> Int) y = id y;
                                 id3 :: (Int -> Int) x = id2 x;
                                 two :: (Bool -> Int -> Int) x y = one y x;
                                 one :: (Int -> Bool -> Int) x y = id x;
                          in id2;
